### ISSUE TITLE
AWS::NoValue in conditional AvailabilityZones triggers errors

### ISSUE THREAD

**Issue by altaurog** [aws-cloudformation/cfn-lint#2839]
### CloudFormation Lint Version

cfn-lint 0.78.2

### What operating system are you using?

Debian 5.10.158-2

### Describe the bug

I have the following in my template for an `AWS::RDS::DBInstance` resource:
```yaml
      MultiAZ: !If ["IsProduction", true, false]
      AvailabilityZone: !If
        - IsProduction
        - AWS::NoValue
        - !Select ['0', {"Fn::GetAZs": {"Ref": "AWS::Region"}}]
```
In a production environment, we use a multi-az deployment, for which availability zone should not be set.

This is triggering the following two messages from cfn-lint:
```
E3031 AvailabilityZone contains invalid characters (Pattern: [a-z0-9-]+) at Resources/DatabaseInstance/Properties/AvailabilityZone/Fn::If/1

W3010 Don't hardcode AWS::NoValue for AvailabilityZones
```
This seems like a bug, but perhaps there is another way I should be doing this?

### Expected behavior

It should recognize that `AWS::NoValue` is neither an invalid value nor a hardcoded availability zone, and issue no warnings or errors.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Parameters:
  Environment:
    Description: Environment name
    Type: String
    AllowedValues:
      - test
      - prod

Conditions:
  IsProduction: !Equals
    - !Ref Environment
    - prod

Resources:
  NewDatabaseInstance:
    Type: "AWS::RDS::DBInstance"
    Properties:
      AllocatedStorage: 100
      AvailabilityZone: !If
        - IsProduction
        - AWS::NoValue
        - !Select ['0', {"Fn::GetAZs": {"Ref": "AWS::Region"}}]
      DBInstanceClass: "db.t4g.micro"
      Engine: "postgres"
      MasterUsername: "postgres"
      MasterUserPassword: "{{resolve:secretsmanager:db-password:SecretString:password}}"
      MaxAllocatedStorage: 400
      MultiAZ: !If ["IsProduction", true, false]
      Port: 5432
      PubliclyAccessible: false
      StorageEncrypted: true
      StorageType: "io1"

  RestoredDatabaseInstance:
    Type: "AWS::RDS::DBInstance"
    Properties:
      AvailabilityZone: !If
        - IsProduction
        - AWS::NoValue
        - !Select ['0', {"Fn::GetAZs": {"Ref": "AWS::Region"}}]
      DBInstanceClass: "db.t4g.micro"
      DBSnapshotIdentifier: "my-snapshot"
      MultiAZ: !If ["IsProduction", true, false]
      Port: 5432
      PubliclyAccessible: false
      StorageEncrypted: true
      StorageType: "io1"
```

**Comment by benbridts** [comment]
You need to put a `!Ref` before `AWS::NoValue`, it's a pseudo-parameter, so it behaves like a parameter (you are doing this correctly for `AWS::Region`).

```yaml
      AvailabilityZone: !If
        - IsProduction
        - !Ref "AWS::NoValue"
        - !Select ['0', {"Fn::GetAZs": {"Ref": "AWS::Region"}}]
```
